content/blog/emacs.md

```diff
@@ -0,0 +1,57 @@
+
+One important attraction on the road from plain user to power user is [Org mode][orgmode], which has been part of Emacs for a while. It's one of the most versatile personal information management systems in existence, but it's also one that doesn't make much sense unless you are already familiar with the overall philosophy of Emacs. Moreover, productive use of Org mode requires significant customization. The move from text editing in Emacs to PIM management with Org mode is a step up the ladder that is much smaller than the step to "Emacs Lisp development", and yet it comes with a high reward. Many other Emacs add-ons play a similar role. Mail handling in Emacs, via [notmuch][notmuch] or [mu4e][mu4e] requires more coding than running a standard mail client, but much less than writing your own mail client from scratch. The malleability of Emacs had led to an ecosystem of add-ons that are themselves malleable, and often have malleability as their main advantage compared to conventional silo-type applications. This ecosystem is important for Emacs' attractiveness today. But the seed that got the ecosystem started is out-of-the-box utility.
+
+## Lesson #2: Think carefully about barriers
+
+It's also interesting to compare Emacs (or Smalltalk) to platforms implementing extensibility through a plug-in framework. Web browsers are the best comparison here, because they share the useful-out-of-the-box feature with Emacs. The big difference is that browsers make a clear distinction between customization through a built-in preference panel and writing extensions using external development tools. There is no continuity on the path from plain user to power user. At some point you have to make the decision to become an extension developer, and attack the steep learning curve that comes with it. Moreover, browsers severely restrict what extensions can modify. In fact, users can't even explore the code of the browser itself from the browser, let alone change it. In contrast, most of Emacs' functionality is implemented in the same language as user extensions, and it is fully documented and accessible from inside Emacs.
+
+Note: [Geoffrey Litt's post][browser-extension-blog-post] on browser extensions takes a different point of view, considering browser extensions as a way to modify not browsers but Web applications. That's the starting point of a another direction of investigation: how to add malleability to systems that were *not* designed for it. But that's not my topic here.
```

Comment:
Random aside, since you mention it here: Adding malleability to systems not designed for it is quite hard, but also quite interesting to me personally. 😄 I'm hoping to share some more concrete ideas about this in the future.

Reply:
Looking forward to that!